lib/matplotlib/bezier.py

```diff
@@ -239,15 +239,114 @@ def __init__(self, control_points):
         coeff = [math.factorial(self._N - 1)
                  // (math.factorial(i) * math.factorial(self._N - 1 - i))
                  for i in range(self._N)]
-        self._px = self._cpoints.T * coeff
+        self._px = (self._cpoints.T * coeff).T
 
     def __call__(self, t):
-        return self.point_at_t(t)
+        t = np.array(t)
+        orders_shape = (1,)*t.ndim + self._orders.shape
+        t_shape = t.shape + (1,)  # self._orders.ndim == 1
+        orders = np.reshape(self._orders, orders_shape)
+        rev_orders = np.reshape(self._orders[::-1], orders_shape)
+        t = np.reshape(t, t_shape)
+        return ((1 - t)**rev_orders * t**orders) @ self._px
 
     def point_at_t(self, t):
         """Return the point on the Bezier curve for parameter *t*."""
-        return tuple(
-            self._px @ (((1 - t) ** self._orders)[::-1] * t ** self._orders))
+        return tuple(self(t))
```

Comment:
Since this function is public and returned a tuple before, this code ensures that the public API doesn't change.